cli/src/commands/revert.rs

```diff
@@ -0,0 +1,204 @@
+                let commit_description = output.into_string_lossy();
+                (commit, commit_description)
+            })
+            .collect_vec()
+    };
+    let mut tx = workspace_command.start_transaction();
+    let original_parent_commit_ids: HashSet<_> = new_parents.iter().ids().cloned().collect();
+    let mut new_base_tree = merge_commit_trees(tx.repo(), &new_parents)?;
+    let mut parent_ids = new_parents.iter().ids().cloned().collect_vec();
+
+    let mut reverted_commits = vec![];
+    for (commit_to_revert, new_commit_description) in
+        &commits_to_revert_with_new_commit_descriptions
+    {
+        let old_base_tree = commit_to_revert.parent_tree(tx.repo())?;
+        let old_tree = commit_to_revert.tree()?;
+        let new_tree = new_base_tree.merge(&old_tree, &old_base_tree)?;
+        let new_parent_ids = parent_ids.clone();
+        let new_commit = tx
+            .repo_mut()
+            .new_commit(new_parent_ids, new_tree.id())
+            .set_description(new_commit_description)
+            .write()?;
+        parent_ids = vec![new_commit.id().clone()];
+        reverted_commits.push(new_commit);
+        new_base_tree = new_tree;
+    }
+
+    // Rebase new children onto the reverted commit.
+    let new_head_ids: Vec<_> = parent_ids;
+    let children_commit_ids_set: HashSet<CommitId> = new_children.iter().ids().cloned().collect();
+    let mut num_rebased = 0;
+    tx.repo_mut().transform_descendants(
+        children_commit_ids_set.iter().cloned().collect_vec(),
+        |mut rewriter| {
+            if children_commit_ids_set.contains(rewriter.old_commit().id()) {
+                let mut child_new_parent_ids = IndexSet::new();
+                for old_parent_id in rewriter.old_commit().parent_ids() {
+                    // If the original parents of the new children are the new parents of
+                    // `target_head_ids`, replace them with `new_head_ids` since we are
+                    // "inserting" the new commits in between the new parents and the new
+                    // children.
+                    if original_parent_commit_ids.contains(old_parent_id) {
+                        child_new_parent_ids.extend(new_head_ids.clone());
+                    } else {
+                        child_new_parent_ids.insert(old_parent_id.clone());
+                    }
+                }
+                // If not already present, add `new_head_ids` as parents of the new child
+                // commit.
+                child_new_parent_ids.extend(new_head_ids.clone());
+                rewriter.set_new_parents(child_new_parent_ids.into_iter().collect());
+            }
+            num_rebased += 1;
+            rewriter.rebase()?.write()?;
+            Ok(())
+        },
+    )?;
+
+    if let Some(mut formatter) = ui.status_formatter() {
+        writeln!(formatter, "Reverted {} commits", reverted_commits.len())?;
+        writeln!(formatter, "Created the following commits:")?;
```

Comment:
nit: maybe these lines can be combined? "Reverted N commits as follows:" or something like that.